### PR TITLE
feat(sdk): extend function with timeout property 

### PR DESCRIPTION
**Pull request by RaphaelManke**
Closes: https://github.com/winglang/wing/issues/1204

This PR adds a timeout property to the `cloud.Function` class so that the execution time of a function can be limited. 
This property is defined in the rfc for the Function class. 
The RFC specifies a default value for timeout of 1 minute.

This PR adds a implementation for the timeout property for the simulator and aws function.

In the AWS function implementation the value of the timeout property is directly mapped to the cdktf property of the aws lambda function construct property.

The simulator implementation uses the node [vm options](https://nodejs.org/docs/latest-v16.x/api/vm.html#vmrunincontextcode-contextifiedobject-options) which offer a timeout property to cancel executions after a defined timeout. 

An alternative implementation of the timeout for the simulator has been not been evalutated because the implemented version resulted in the expected behaviour with a minimal effort. 
Side efects are not known yet. 
In case of a timeout in the simulator the User gets informed that a script execution has timed out.
This message could be improved to give the user feedback which inflight function has timed out.

*By submitting this pull request, I confirm that my contribution is made under the terms of the 
[Monada Contribution License](https://docs.winglang.io/terms-and-policies/contribution-license.html)*.

### CODE REVIEW

**Review by Chriscbr — COMMENTED** [review]
I think this looks good! Maybe for the simulator target we can throw an error if someone specifies a `timeout` value?

File: libs/wingsdk/src/target-sim/function.inflight.ts [hunk 1]
```diff
@@ -75,7 +77,7 @@ export class Function implements IFunctionClient, ISimulatorResourceInstance {
     return this.context.withTrace({
       message: `Invoke (payload="${JSON.stringify(payload)}").`,
       activity: async () => {
-        return vm.runInContext(wrapper, context);
+        return vm.runInContext(wrapper, context, { timeout: this.timeout });
```
Comment (Chriscbr) [inline libs/wingsdk/src/target-sim/function.inflight.ts:80]: Aha, so there's a timeout option! Brilliant :)

Comment (RaphaelManke) [reply libs/wingsdk/src/target-sim/function.inflight.ts:80]: yes, I thought that would be clean solution, to make sure the script execution is really stopped.

**Comment by RaphaelManke** [comment]
> Looks like there's a build error related to [this Wing example](https://github.com/winglang/wing/blob/main/examples/tests/invalid/struct_expansion.w) with this line in the logs:
> 
> ```
> wingc: thread 'sanity::cannot_compile_invalid_files' panicked at 'Not all fields of FunctionProps (at src/cloud/function.ts:21:1) are initialized.', libs/wingc/src/type_check.rs:902:21
> ```
> 
> What's weird is that `timeout` is an optional property, so I think the correct behavior is for the code to compile successfully. This is likely a bug in the compiler. You're welcome to dig into the compiler code if you'd like, but I think it's also totally okay to just update the example to specify the `timeout` for now, and someone can fix the error in a separate PR. 🙂

I am not sure how to fix the example. 
How do I test this build locally?
How do I use the Duration class from the std lib to set a timeout? 
what I'd like to do is 
```rust
new cloud.Function(
  handler, 
  cloud.FunctionProps {
    env: Map<str> {}
    timeout: std.Duration.fromMinutes(1)
  }
);

```

**Comment by Chriscbr** [comment]
@RaphaelManke To represent a duration in Wing, you can write `1s`, `1m`, or `1h` (with any numbers)

To run all of the tests you can follow the instructions here: https://docs.winglang.io/contributors/handbook#-how-do-i-run-e2e-tests

**Review by Chriscbr — APPROVED** [review]

**Comment by Chriscbr** [comment]
Thanks @RaphaelManke for contributing this! 🙏

**Comment by staycoolcall911** [comment]
Congrats on your first PR @RaphaelManke 👏

**Pull request merged**